azure-pipelines-pr-validation.yml

```diff
@@ -42,238 +46,296 @@ variables:
-      displayName: Perform Cleanup Tasks
-      condition: succeededOrFailed()
-
-- stage: insert
-  displayName: Create Insertion
-  dependsOn:
-    - build
-
-  jobs:
-  - job: insert
-    displayName: Insert to VS
-    pool:
-      vmImage: windows-2019
-    steps:
-    - powershell: Write-Host "##vso[task.setvariable variable=SourceBranchName]$('$(Build.SourceBranch)'.Substring('refs/heads/'.Length))"
-      displayName: Setting SourceBranchName variable
-      condition: succeeded()
-
-    - template: eng/pipelines/insert.yml
+    customBuildTags:
+    - ES365AIMigrationTooling
+    stages:
+
+    - stage: build
+      displayName: Build and Test
+
+      jobs:
+      - job: PRValidationBuild
+        displayName: PR Validation Build
+        timeoutInMinutes: 360
+        templateContext:
+          outputs:
+
+          # Publish OptProf configuration files to the artifact service
+          - output: artifactsDrop
+            displayName: 'OptProf - Publish to Artifact Services - ProfilingInputs'
+            condition: succeeded()
+            dropServiceURI: 'https://devdiv.artifacts.visualstudio.com'
+            buildNumber: 'ProfilingInputs/$(System.TeamProject)/$(Build.Repository.Name)/$(SourceBranchName)/$(OriginalBuildNumber)'
+            sourcePath: '$(Build.SourcesDirectory)\artifacts\OptProf\$(BuildConfiguration)\Data'
+            toLowerCase: false
+            usePat: true
+            accessToken: $(_DevDivDropAccessToken)
+            retentionDays: 90
+
+          # Publish OptProf generated JSON files as a pipeline artifact. This allows for easy inspection from
+          # a build execution.
+          - output: pipelineArtifact
+            displayName: 'Publish OptProf Data Files'
+            condition: succeeded()
+            targetPath: '$(Build.SourcesDirectory)\artifacts\OptProf\$(BuildConfiguration)\Data'
+            artifactName: 'OptProf Data Files'
+
+          - output: pipelineArtifact
+            displayName: 'Publish Logs'
+            condition: succeededOrFailed()
+            targetPath: '$(Build.SourcesDirectory)\artifacts\log\$(BuildConfiguration)'
+            artifactName: 'Build Diagnostic Files'
+            publishLocation: Container
+
+          - output: pipelineArtifact
+            displayName: 'Publish Ngen Logs'
+            condition: succeeded()
+            targetPath: '$(Build.SourcesDirectory)\artifacts\log\$(BuildConfiguration)\ngen'
+            artifactName: 'NGen Logs'
+            publishLocation: Container
+
+          # Publishes setup VSIXes to a drop.
+          # Note: The insertion tool looks for the display name of this task in the logs.
+          - output: microBuildVstsDrop
+            displayName: Upload VSTS Drop
+            condition: succeeded()
+            dropFolder: 'artifacts\VSSetup\$(BuildConfiguration)\Insertion'
+            dropName: $(VisualStudio.DropName)
+            accessToken: $(_DevDivDropAccessToken)
```

Comment:
You might want to set expiration date for this like https://github.com/dotnet/roslyn/pull/73190

Reply:
Good call. I want the pr-val pipeline to be as similar as possible to the official build.